### PR TITLE
Implement CardView for MultipleIsbn reposnses

### PR DESCRIPTION
**Pull request by TheFlash98**
Implemented a CardView using RecyclerView which shows the book photo, title, author name and a short description. In case the image URL is missing, a icon is loaded in place of the image. 
Thanks to @nihal111 for helping with loading images.

### CODE REVIEW

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class MultipleIsbnActivity extends AppCompatActivity {
+
+    public static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_recycler);
+
+        RecyclerView rv = (RecyclerView) findViewById(R.id.rv);
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java]: `rv` is a terrible variable name. Reminds me of [this scene from the movie Ta Ra Rum Pum](https://www.youtube.com/watch?v=mnRkM_TwACc).
Lets use `recyvlerView` instead

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import com.iitb.ferozepurwale.lendit_app.models.Book;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+class RVAdapter extends RecyclerView.Adapter<RVAdapter.BookViewHolder> {
+
+    private Context mContext;
+    private List<Book> books;
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java]: See, access specifiers.

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    public void onAttachedToRecyclerView(RecyclerView recyclerView) {
+        super.onAttachedToRecyclerView(recyclerView);
+    }
+
+    public static class BookViewHolder extends RecyclerView.ViewHolder {
+        CardView cv;
+        TextView bookName;
+        TextView bookAuthor;
+        ImageView bookPhoto;
+        TextView bookDes;
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java]: Access Specifiers.

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public void onBindViewHolder(final BookViewHolder bookViewHolder, int i) {
+        bookViewHolder.bookName.setText(books.get(i).title);
+        bookViewHolder.bookAuthor.setText(books.get(i).author);
+
+        String imageUrl = books.get(i).image;
+        if (!imageUrl.equals("")) {
+            Picasso.with(mContext).load(imageUrl).into(bookViewHolder.bookPhoto);
+        } else {
+            //todo - implement a default image in case img_url is indeed empty
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java]: Remove the todo comment

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            bookViewHolder.bookPhoto.setImageResource(R.drawable.ic_book_loading);
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java]: This else if for an image when there is no image url specified. Not one for loading.
On second thoughts, add a loading image to be displayed before doing a fetch from the imageUrl.

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        rv.setHasFixedSize(true);
+
+        LinearLayoutManager llm = new LinearLayoutManager(getApplicationContext());
```
[the remark this answers is not in the document]
Comment (nihal111) [reply app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java]: Also `llm` is a terrible variable name. Please use `linearLayoutManager`

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                +"\"description\": \"Traces the unlikely friendship of a wealthy Afghan youth and a servant's son, in a tale that spans the final days of Afghanistan's monarchy through the atrocities of the present day.\","
+                +"\"rating\": 4.0}";
+        Gson gson = new Gson();
+        Book book = gson.fromJson(jsonString, Book.class);
+        List<Book> books = new ArrayList<>();
+        books.add(book);
+        books.add(book);
+        books.add(book);
+        System.out.println(books.get(1).title);
+        RVAdapter adapter = new RVAdapter(getApplicationContext(), books);
```
[the remark this answers is not in the document]
Comment (nihal111) [reply app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java]: Also `RVAdapter` -> `RecyclerViewAdapter`

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.iitb.ferozepurwale.lendit_app.models.Book;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+class RVAdapter extends RecyclerView.Adapter<RVAdapter.BookViewHolder> {
```
Comment (nihal111) [inline app/src/main/java/com/iitb/ferozepurwale/lendit_app/RVAdapter.java]: rename to something like BooksRecyclerViewAdapter

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (TheFlash98) [reply app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java]: Not letting me set `private` or `public` to `LinearLayoutManager`

File: app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (TheFlash98) [reply app/src/main/java/com/iitb/ferozepurwale/lendit_app/MultipleIsbnActivity.java]: Not letting me give access specifiers again.

**Pull request merged**